### PR TITLE
Also try keyserver.ubuntu.com for MySQL key

### PR DESCRIPTION
**Pull request by martijnvans**
This is to help with the occasional flake where pgp.mit.edu is inaccessible.

### CODE REVIEW

File: integration_test/third_party_apps_data/applications/mysql/debian_ubuntu/install [hunk 1]
```diff
@@ -4,7 +4,7 @@ sudo apt update
 sudo apt install -y wget
 
 # From https://dev.mysql.com/doc/mysql-apt-repo-quick-guide/en/#repo-qg-apt-repo-manual-setup
-sudo apt-key adv --keyserver pgp.mit.edu --recv-keys 3A79BD29
+sudo apt-key adv --keyserver pool.sks-keyservers.net --recv-keys 3A79BD29
```
Comment (igorpeshansky) [inline integration_test/third_party_apps_data/applications/mysql/debian_ubuntu/install:7]: Maybe use keyserver.ubuntu.com instead? See https://github.com/rvm/rvm/issues/5096.

**Review by igorpeshansky — CHANGES_REQUESTED** [review]

File: integration_test/third_party_apps_data/applications/mysql/debian_ubuntu/install (hunk 1 above)
Comment (martijnvans) [reply integration_test/third_party_apps_data/applications/mysql/debian_ubuntu/install]: done. the latest snapshot tries both locations. sadface

File: integration_test/third_party_apps_data/applications/mysql/debian_ubuntu/install [hunk 2]
```diff
@@ -3,8 +3,11 @@ set -e
 sudo apt update
 sudo apt install -y wget
 
-# From https://dev.mysql.com/doc/mysql-apt-repo-quick-guide/en/#repo-qg-apt-repo-manual-setup
-sudo apt-key adv --keyserver pgp.mit.edu --recv-keys 3A79BD29
+function install_key() {
```
Comment (igorpeshansky) [inline integration_test/third_party_apps_data/applications/mysql/debian_ubuntu/install:6]: [Optional] `add_apt_key`?

**Review by igorpeshansky — CHANGES_REQUESTED** [review]

**Review by jefferbrecht — APPROVED** [review]

File: integration_test/third_party_apps_data/applications/mysql/debian_ubuntu/install (hunk 2 above)
Comment (martijnvans) [reply integration_test/third_party_apps_data/applications/mysql/debian_ubuntu/install]: done

**Review by igorpeshansky — APPROVED** [review]
LGTM :shipit:

**Comment by quentinmit** [comment]
Honestly you should just switch wholesale; `pgp.mit.edu` is very very flaky.

(Or better, you should import the key from an https URL instead of using a keyserver)

**Comment by martijnvans** [comment]
Quentin, I'm not having luck finding such a URL and I'm not sure how to pass one in even if I found it. I feel like progress on this PR has stalled and I could use some more help from you to get it going again, or I could merge it as is and admit that it's not the best possible implementation and we can fix it later if needed. Which would you prefer?

**Comment by quentinmit** [comment]
> Quentin, I'm not having luck finding such a URL and I'm not sure how to pass one in even if I found it. I feel like progress on this PR has stalled and I could use some more help from you to get it going again, or I could merge it as is and admit that it's not the best possible implementation and we can fix it later if needed. Which would you prefer?

The first bullet point at https://dev.mysql.com/doc/mysql-apt-repo-quick-guide/en/#repo-qg-apt-repo-manual-setup says

> Download the MySQL GPG Public key (see [Signature Checking Using GnuPG](https://dev.mysql.com/doc/refman/8.0/en/checking-gpg-signature.html) on how to do that) and save it to a file

I agree that I can't seem to find that as a separate file directly... one option would be

```
curl https://dev.mysql.com/doc/refman/8.0/en/checking-gpg-signature.html | perl -nle 's/^.*(-----BEGIN)/\1/; print if /BEGIN PGP PUBLIC KEY BLOCK/../END PGP PUBLIC KEY BLOCK/'
```

Another option appears to be to just install https://dev.mysql.com/get/mysql-apt-config_0.8.22-1_all.deb which will configure the apt repo and load the appropriate key.

**Comment by jefferbrecht** [comment]
Tests all passed and it's up to date with master. This is a substantial improvement that'll help avoid many test failures so I'll go ahead and merge it now. If there are any outstanding threads of discussion here then a follow-up PR would be appropriate.

**Comment by quentinmit** [comment]
> Tests all passed and it's up to date with master. This is a substantial improvement that'll help avoid many test failures so I'll go ahead and merge it now. If there are any outstanding threads of discussion here then a follow-up PR would be appropriate.

Yes, there are two suggestions above for making this more reliable that are waiting for reply.

**Comment by martijnvans** [comment]
I'm taking those suggestions under consideration in https://github.com/GoogleCloudPlatform/ops-agent/pull/510.

**Pull request merged**